pallets/xcm-core-buyer/src/lib.rs

```diff
@@ -506,17 +597,45 @@ pub mod pallet {
                 if pending_blocks_ttl < <frame_system::Pallet<T>>::block_number() {
                     PendingBlocks::<T>::remove(para_id);
                 } else {
-                    return Err(Error::<T>::BlockProductionPending.into());
+                    return Err(BuyingError::BlockProductionPending {
+                        ttl: pending_blocks_ttl,
+                        current_block_number: <frame_system::Pallet<T>>::block_number(),
+                    });
                 }
             }
 
             // Check that the para id is a parathread
-            let parathread_params = T::GetParathreadParams::get_parathread_params(para_id);
-            if parathread_params.is_none() {
-                return Err(Error::<T>::NotAParathread.into());
+            let maybe_parathread_params = T::GetParathreadParams::get_parathread_params(para_id);
+            let parathread_params = if let Some(parathread_params) = maybe_parathread_params {
+                parathread_params
+            } else {
+                return Err(BuyingError::NotAParathread);
+            };
+
+            let maybe_latest_author_info =
+                T::LatestAuthorInfoFetcher::get_latest_author_info(para_id);
+            if let Some(latest_author_info) = maybe_latest_author_info {
+                let current_slot = T::SlotBeacon::slot();
+                if !parathread_params.slot_frequency.should_parathread_buy_core(
+                    Slot::from(current_slot as u64),
+                    Slot::from(2u64),
```

Comment:
```suggestion
                    Slot::from(2u64),
```

this should be a config parameter probably no?

Reply:
yes. I have added to-do for that.